picard/log.py

```diff
@@ -100,7 +112,11 @@ def __init__(self, maxlen):
 
     def contents(self, prev=-1):
         with self._queue_lock:
-            contents = [x for x in self._log_queue if x.pos > prev]
+            # If log queue is empty, return
+            if len(self._log_queue) == 0:
+                return []
+            bounds = _calculate_bounds(prev, self._log_queue[0].pos, self._log_queue[-1].pos, len(self._log_queue))
+            contents = [self._log_queue[i] for i in range(*bounds)]
```

Comment:
I think we could make this an iterator:

    (self._log_queue[i] for i in range(*bounds))

`contents` only gets used to iterate over it afterwards, so we don't really need to create a list here.